src/EditorFeatures/CSharpTest/MoveStaticMembers/CSharpMoveStaticMembersTests.cs

```diff
@@ -0,0 +1,1439 @@
+namespace TestNs1
+{
+    public class Class1
+    {
+        public static int Test[||]MethodInt()
+        {
+            return 0;
+        }
+
+        public static bool TestMethodBool()
+        {
+            return false;
+        }
+    }
+}
+        </Document>
+    </Project>
+</Workspace>";
+            var selectedDestinationName = "Class1Helpers";
+            var newFileName = "Class1Helpers.cs";
+            var selectedMembers = ImmutableArray.Create("TestMethodBool");
+            var expectedResult = @"
+<Workspace>
+    <Project Language=""C#"" AssemblyName=""CSAssembly1"" CommonReferences=""true"">
+        <Document FilePath=""Class1.cs"">
+namespace TestNs1
+{
+    public class Class1
+    {
+        public static int TestMethodInt()
+        {
+            return 0;
+        }
+    }
+}
+        </Document>
+        <Document FilePath=""Class1Helpers.cs"">namespace TestNs1
+{
+    static class Class1Helpers
+    {
+
+        public static bool TestMethodBool()
+        {
+            return false;
+        }
+    }
+}</Document>
+    </Project>
+</Workspace>";
+            await TestMovementAsync(initialMarkup,
+                expectedResult,
+                selectedDestinationName,
+                selectedMembers,
+                newFileName).ConfigureAwait(false);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMoveStaticMembers)]
+        public async Task TestMoveOneOfEach()
+        {
+            // Bug where PullMembersUp doesn't apply formatting annotation
+            // When the destination ends up with more than one member:
+            // https://github.com/dotnet/roslyn/issues/54847
+            // so we keep the weird formatting for now
+            var initialMarkup = @"
+<Workspace>
+    <Project Language=""C#"" AssemblyName=""CSAssembly1"" CommonReferences=""true"">
+        <Document FilePath=""Class1.cs"">
+using System;
+
+namespace TestNs1
+{
+    public class Class1
+    {
+        public static int Test[||]Method()
+        {
+            return 0;
+        }
+
+        public static bool TestProperty { get; set; }
+
+        public static int TestField;
+
+        public static event EventHandler TestEvent;
+    }
+}
+        </Document>
+    </Project>
+</Workspace>";
+            var selectedDestinationName = "Class1Helpers";
+            var newFileName = "Class1Helpers.cs";
+            var selectedMembers = ImmutableArray.Create(
+                "TestMethod",
+                "TestField",
+                "TestProperty",
+                "TestEvent");
+            var expectedResult = @"
+<Workspace>
+    <Project Language=""C#"" AssemblyName=""CSAssembly1"" CommonReferences=""true"">
+        <Document FilePath=""Class1.cs"">
+using System;
+
+namespace TestNs1
+{
+    public class Class1
+    {
+    }
+}
+        </Document>
+        <Document FilePath=""Class1Helpers.cs"">using System;
+
+namespace TestNs1
+{
+    static class Class1Helpers
+    {
+
```

Comment:
We should open an issue for the leading whitespace in these classes.

Reply:
I believe it is an extension of #54847. We can add a formatter annotation to the new class in this case, but this will still be an issue when moving to existing types